fix: handle undefined metadata.name access to prevent TypeError across UI components

Bvvvp009 opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

This PR fixes `TypeError: Cannot read properties of undefined (reading 'name')` that occurs when opening MetaMask, particularly in Opera browser. The error happens when UI components attempt to access `account.metadata.name` but `account.metadata` is `undefined` or `null` during wallet unlock.

The fix applies optional chaining (`?.`) and nullish coalescing (`|| ''`) to all `metadata.name` access patterns across 50+ UI components, controllers, and selectors. This ensures graceful handling of undefined metadata without breaking existing functionality.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/35659?quickstart=1)

## **Changelog**

CHANGELOG entry: Fixed a bug that was causing MetaMask to crash with "Cannot read properties of undefined (reading 'name')" error when opening the extension, particularly in Opera browser.

## **Related issues**

Fixes: TypeError: Cannot read properties of undefined (reading 'name') error in Opera browser

## **Manual testing steps**

1. Open MetaMask extension in Opera browser
2. Try to unlock the wallet
3. Verify no TypeError occurs during unlock process
4. Test account switching and navigation to ensure all account names display correctly
5. Verify the fix works across different account states (new accounts, imported accounts, etc.)

## **Screenshots/Recordings**

<img width="525" height="849" alt="Screenshot 2025-09-04 174616" src="https://github.com/user-attachments/assets/d6c07668-7804-4217-8443-90b81b6c2b12" />

<img width="927" height="547" alt="Screenshot 2025-09-04 175038" src="https://github.com/user-attachments/assets/a7999b1d-25f2-4a11-b66e-b487cf7da09d" />

### **Before**

Error occurs when opening MetaMask:

### **After**

MetaMask opens without errors, account names display correctly even with incomplete metadata.

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I've documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I've applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Safely access account/wallet `metadata.name` throughout the app using optional chaining with fallbacks (empty string or address) to prevent runtime errors.
> 
> - **Controllers/Lib**:
>   - Use `metadata?.name` with fallbacks in `PreferencesController` identities sync, `AccountIdentitiesPetnamesBridge`, and `metamask-controller` (entropy sources).
>   - Update `getUniqueAccountName` to check `account.metadata?.name`.
> - **UI Components**:
>   - Replace direct `metadata.name` reads with `metadata?.name ?? address` (or `''` where appropriate) across account pickers, alerts, account list items, QR code, account details, wallet details, permission modal, and private key views.
>   - Adjust multichain account/wallet list to safely read `wallet.metadata?.name`, `groups`, and `type`.
> - **Selectors**:
>   - Guard name lookups with `metadata?.name` in bridge selectors and core selectors (e.g., address-to-name maps, display names).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 2aacd2822bbdc13d37c59462414c548b94255b9d. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

Bvvvp009 commented:
I have read the CLA Document and I hereby sign the CLA

Bvvvp009 commented:
bugbot run

Bvvvp009 reviewed (commented):
Resolved this issue.

owencraston reviewed (changes requested):
This is not a valid fix for this issue. Sure it will remove the error but then the UI will have no account name. We need to properly handle this null case on a case by case basis.

Bvvvp009 committed:
fix(metadata): handle null metadata.name cases with proper fallbacks

- Fix UI components to use address fallback (?? address) for meaningful display
- Fix non-UI components to use empty string fallback (|| '') for APIs/logging
- Standardize metadata access pattern to ?.metadata?.name across codebase
- Addresses code owner feedback on case-by-case null handling
- Prevents Opera crash: Cannot read properties of undefined (reading 'name')

Fixes: #35659

Bvvvp009 committed:
fix(metadata): add missing type assertion for walletData

Bvvvp009 committed:
fix(formatting): break long line in connected-sites container

Bvvvp009 committed:
chore: remove accidentally committed tatus file

- Remove development artifact that was accidentally committed
- File contained git branch listing output
- Not part of the actual codebase

Bvvvp009 committed:
Merge branch 'main' into main

Bvvvp009 committed:
Merge branch 'main' into main

Bvvvp009 committed:
Merge branch 'main' into main

Bvvvp009 committed:
Merge branch 'main' into main

Bvvvp009 committed:
Merge branch 'main' into main

Bvvvp009 committed:
Merge branch 'main' into main
